datalad/support/gitrepo.py

```diff
@@ -1051,7 +1055,7 @@ def get_active_branch(self):
         except TypeError as e:
             if "HEAD is a detached symbolic reference" in str(e):
                 lgr.debug("detached HEAD in {0}".format(self))
-                return None, None
+                return None
```

Comment:
I believe that this double `None` was done on purpose to distinguish case of having no branch (e.g. new repo, no commit yet) from having a detached branch 

remainders:
```
datalad/distribution/get.py-    ds.repo.update_submodule(sm_path, init=True)
datalad/distribution/get.py-    updated_branch = subrepo.get_active_branch()
datalad/distribution/get.py:    if branch and (not updated_branch or updated_branch == (None, None)):
--
datalad/distribution/update.py-        # handle merge in plain git
datalad/distribution/update.py-        active_branch = repo.get_active_branch()
datalad/distribution/update.py:        if active_branch == (None, None):
```